src/molecules/Tabs/Tabs.native.js

```diff
@@ -0,0 +1,134 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Dimensions } from 'react-native';
+import PropTypes from 'prop-types';
+import { TabView } from 'react-native-tab-view';
+import { getIconNames } from '../../_helpers/icon';
+import isDefined from '../../_helpers/isDefined';
+import TabBar from './TabBar';
+
+const initialLayout = { width: Dimensions.get('window').width };
+
+const Tab = () => {
+  return null;
+};
+
+Tab.propTypes = {
+  label: PropTypes.string.isRequired,
+  onPress: PropTypes.func,
+  disabled: PropTypes.bool,
+  icon: PropTypes.oneOf(getIconNames()),
+  testID: PropTypes.string,
+};
+
+Tab.defaultProps = {
+  onPress: () => {},
+  disabled: false,
+  testID: 'ds-tab-button',
+};
+
+const getTabs = ({ children }) => {
+  // Return only Tab Elements
+  return React.Children.toArray(children).filter(
+    (child) => child.type.name && child.type.name === 'Tab',
+  );
+};
+
+const shouldDisableSwipe = ({ tabs }) => {
+  let disableSwipe = false;
+  // Disable swipe if any of the tab is disabled
+  tabs.forEach((tab) => {
```

Comment:
you can just use `find`. It'll be optimised since it exits as soon as the first matching element is found contrast to forEach which will keep on iterating till the last element.